### PR TITLE
Package with Nix

### PR DESCRIPTION
**Pull request by mausch**
Hi and thanks for this Boulder Dash clone!
Here's a Nix flake so it can be easily run.
Not sure if you're familiar with Nix? Otherwise I can explain.

### CODE REVIEW

**Comment by rossberg** [comment]
Thanks, yes, I've worked on a project that uses nix. This looks good, but is there a way to support it without cluttering the root dir with two more files, e.g., perhaps by putting them into some subdirectory? I prefer tidy directories without distractions, and already dislike that dune litters it with two files. :)

**Comment by mausch** [comment]
It's possible yeah, users would have to run something like `nix run github:rossberg/boulder-dash?dir=nix` which is kinda unusual.
I do see what you mean about tidy directories! What most projects do IME is move the source code to a subdirectory and leave project files like flake, Makefile and dune stuff in the root. Some examples:

https://github.com/mirage/ocaml-cohttp
https://github.com/ocaml-ppx/ppx_deriving
https://github.com/andrejbauer/plzoo
https://github.com/matijapretnar/eff

**Comment by rossberg** [comment]
Great, then I think I'd rather use a nix/ directory. Happy to add the instruction to the README if necessary. The src/ convention is overkill for a small repo that contains nothing but source.

**Comment by mausch** [comment]
Done :+1: 

Until this gets merged you can try it with: 
```
nix run github:rossberg/boulder-dash/2b90b4f7e65df014e0697d5748ba11fabd3688ec?dir=nix
```

**Comment by rossberg** [comment]
Great, thanks!

Small aside: please never force-push to a GH branch under review, as it breaks basic functionality of Github's review workflow and UI. ;)

File: render.ml [hunk 1]
```diff
@@ -70,7 +70,7 @@ let deinit () =
 
 let _ = preinit ()  (* has to run before Graphics.make_image *)
 
-let bmp = Bmp.load "sprites.bmp" 1
+let bmp = Bmp.load (Filename.concat (Filename.dirname Sys.argv.(0)) "sprites.bmp") 1
```
Comment (rossberg) [inline render.ml]:
```suggestion
let bmp = Bmp.load Filename.(concat (dirname Sys.argv.(0)) "sprites.bmp") 1
```

**Review by rossberg — APPROVED** [review]

**Comment by rossberg** [comment]
@mausch, I tried to keep the nix files updated with version 2, but my nix is super-rudimentary and I have no way of testing them. Also, they only cater for the graphics build, since they lack dependencies on tsdl and raylib.

What would be the best way to (optionally!) support the other backends? Make it three separate packages? How does that work?

**Comment by mausch** [comment]
Yep since they're alternative backends they'd be represented as separate packages. 
One of them should still be the `default` one, possibly the most stable backend or the one with the widest compatibility.

**Comment by rossberg** [comment]
Can I interest you in contributing that? After browsing nix docs for a while, I'm still non the wiser how to express multiple packages. Especially when I cannot test it.

**Pull request merged**